utils/id_hash.go

```diff
@@ -1,4 +1,4 @@
-package keys
+package utils
```

Comment:
`utils` module was already there, I just added a file to it. I could have put it in `./util.go`, but functions in there cannot be reusable by all module (cyclic-dependency). These functions are not used by anything else at the moment, but I thought it would be nice to have that ability.

Reply:
That's because I did not realize you had created an additional module on a previous PR. Note that catch-all util modules are even officially discouraged: https://blog.golang.org/package-names